base/applications/mspaint/winproc.cpp

```diff
@@ -157,6 +157,51 @@ void CMainWindow::InsertSelectionFromHBITMAP(HBITMAP bitmap, HWND window)
     ForceRefreshSelectionContents();
 }
 
+LRESULT CMainWindow::OnMouseWheel(UINT nMsg, WPARAM wParam, LPARAM lParam, BOOL& bHandled)
+{
+    INT xPos = GET_X_LPARAM(lParam), yPos = GET_Y_LPARAM(lParam);
+    INT zDelta = (SHORT)HIWORD(wParam);
+    //UINT fwKeys = LOWORD(wParam);
+
+    if (::GetAsyncKeyState(VK_CONTROL) < 0)
+    {
+        if (zDelta < 0)
+        {
+            if (toolsModel.GetZoom() > MIN_ZOOM)
+                zoomTo(toolsModel.GetZoom() / 2, xPos, yPos);
+        }
+        else if (zDelta > 0)
+        {
+            if (toolsModel.GetZoom() < MAX_ZOOM)
+                zoomTo(toolsModel.GetZoom() * 2, xPos, yPos);
+        }
+    }
+    else
+    {
+#define SCROLL_COUNT 5
+        for (INT i = 0; i < SCROLL_COUNT; ++i)
```

Comment:
What exactly does SCROLL_COUNT do?
If it is the amount by which to scroll, isn't there a configurable system setting for that, as well?
Also, the pre-processor define here looks a bit awkward.

Reply:
One message had not enough effective in the scroll amount.
OK, now it uses sys params.

Reply:
Windows 10 Paint does support Ctrl+Wheel and Plain Wheel but doesn't support Shift+Wheel.

Reply:
In that context Shift+Wheel looks like a sensible minor addition.